### DISCUSSION TITLE
Calling external api using JsonClient

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#7301] [Help] [answered]
Hi All,
Please am using `JsonClient ` from serenity endpoint to call an external api as below:
it works perfectly except that Serenity doesnt wait for the external call to be completed before executing other codes in the endpoint class.
Even if I use `ServiceResponse`, its still the same
Please is there something I am missing here.

```
var client = new JsonServiceClient("http://app.boxing.com");
var response = client.Post<ClientResponse>("api/econnect/PostVendor", new Econnect { ReqId = Convert.ToInt32(request.RequestId), ReqType = "VEND" });

if (response.Success)
{
    Console.WriteLine("successful: " + response.Message);
}

public class ClientResponse
   {
       public bool Success { get; set; }
       public string Message { get; set; }
   }

```

**Comment by rhemasystems** [comment]
Please any help will be greatly appreciated @minhhungit

**Reply by minhhungit** [reply, marked as answer]
@rhemasystems try to use .net core HttpClient 

var response = await _httpClient.PostAsync(url, content);
